mantle/kola/cluster/cluster.go

```diff
@@ -139,20 +139,15 @@ func DropFile(machines []platform.Machine, localPath string) error {
 func (t *TestCluster) SSH(m platform.Machine, cmd string) ([]byte, error) {
 	var stdout, stderr []byte
 	var err error
-	sshCompleted := make(chan bool)
-	go func() {
+	f := func() {
 		stdout, stderr, err = m.SSH(cmd)
-		sshCompleted <- true
-	}()
-
-	select {
-	case <-t.H.TimeoutContext.Done():
-		t.H.FailNow()
-	case <-sshCompleted:
-		if len(stderr) > 0 {
-			for _, line := range strings.Split(string(stderr), "\n") {
-				t.Log(line)
-			}
+	}
+
+	errMsg := fmt.Sprintf("ssh: %s", cmd)
+	t.H.RunWithExecTimeoutCheck(f, errMsg)
```

Comment:
Can we add some context here in a comment about this fucntion being able to break us out of the goroutine if the time expires.